src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -100,7 +100,11 @@
                         ;; Try to get a connection to make sure the db is ready
                         (.close (.getConnection datasource))
                         (when-let [{:keys [migration-db migration-dir replication-mode]} migration-opts]
-                          (when-not (= replication-mode :replica)
+                          (if (= replication-mode :replica)
+                            (loop []
+                              (when (not-empty (migration/uncompleted-migrations migration-db migration-dir))
+                                (Thread/sleep 1500)
```

Comment:
I have no idea what the right number is, I just came up with something I thought was reasonable, and I'm definitely open to suggestions. I don't see a lot of value in a user being able to customize the polling-interval here because the db-timeout in general is customizable.